src/compiler/parser.cc

```diff
@@ -1048,14 +1054,36 @@ Class* Parser::parse_class_interface_monitor_or_mixin(bool is_abstract) {
   } else {
     name = parse_identifier();
     bool requires_super = false;
-    if (current_token() == Token::IDENTIFIER && current_token_data() == Symbols::extends) {
+    if (at(Symbols::extends)) {
       consume();
       requires_super = true;
     }
-    if (current_token() == Token::IDENTIFIER && current_token_data() != Symbols::implements) {
+
+    if (current_token() == Token::IDENTIFIER && !at(Symbols::implements) && !at(Symbols::with)) {
```

Comment:
This feels like a poor use of `at`. I'd just check `current_token_data`.

Reply:
done.